HEN_HOUSE/egs++/ausgab_objects/egs_fluence_scoring/egs_fluence_scoring.h

```diff
@@ -0,0 +1,1074 @@
+
+  /* Fluence Scoring Arrays */
+  EGS_ScoringArray **flu;   // differential fluence: primaries + secondaries
+  EGS_ScoringArray **flu_p; // differential fluence: primaries only 
+  EGS_ScoringArray  *fluT;  // Total fluence: primaries + secondaries
+  EGS_ScoringArray  *fluT_p;// Total fluence: primaries only
+
+  /* Regions flags */
+  vector<bool> is_sensitive;     // flag scoring regions
+  vector<bool> is_source;        // Flag regions such as brems target or radiactive source
+                                 // Interacting particles not subjected to classification
+  vector <int> f_start, f_stop;  // Markers for group regions input
+  vector <int> f_region;         // Input list of scoring regions
+  vector <int> s_region;         // Input list of source regions
+  string       f_regionsString;  // Input string of scoring regions or labels
+  string       s_regionsString;  // Input string of source regions or labels
+  int          n_scoring_regions;// number of scoring regions
+  int          n_source_regions; // number of source regions
+  int          nreg;             // regions in geometry
+  int          max_reg;          // maximum scoring region number
+  int          active_region;    // Region showing calculation progress
+  bool score_in_all_regions;
+  bool source_in_all_regions;
+
+  EGS_Float norm_u;              // User normalization
+
+  /* Energy grid inputs */
+  EGS_Float flu_a, flu_a_i,
+            flu_b,
+            flu_xmin,
+            flu_xmax;
+  int       flu_s,
+            flu_nbin;
+  /* Classification variables */
+  EGS_Float m_primary, 
+            m_tot;
+
+  string  particle_name;
+  /* Auxiliary input variables*/
+  bool    verbose, 
+          score_spe,
+          score_primaries;
+};
+
+/*! \brief Ausgab object for scoring fluence at circular or rectangular fields
+
+  \ingroup AusgabObjects
+
+  A linear track-length estimator in the zero-thickness limit is used to 
+  compute fluence for either a circular field or a rectangular screen of 
+  arbitrary resolution (pixels). Rectangular fields are by default at Z = 0 
+  directed along the positive z-axis. An affine transofrmation can be included
+  to place a rectangular field anywhere in space. User can request to score primary 
+  fluence as well as differential fluence. If fluence for more than one particle type 
+  is desired, multiple AOs are required. 
+  
+  For charged particle fluence scoring this method can be inefficient as 
+  it checks at every single step whether a charged particle is aimed at scoring field.
+  To improve the efficieny for charged particles, one has the option to define contributing regions 
+  from where to score. However, users must be careful to select all regions from where
+  charged particles can cross the scoring field. This option has the added benefit of 
+  allowing to estimate the contribution to the fluence from specific regions in the geometry.
+
```

Comment:
If you use a phase-space source where the latch is already set by BEAMnrc, will it cause any problems? It might be worth noting that this ausgab object overwrites the latch, so any output from egs_phsp_scoring will contain the modified latch. We should probably also have this warning in other apps like egs_chamber..

Reply:
This AO will NOT modify latch. It will take whatever latch the particle has and interpret it accordingly. Phase space files will have primary and secondary particles according to the source application. I don't see a problem there! ;-P

Reply:
@rtownson is this "resolved"?

Reply:
No, currently it still sets latch=1 even if it was already non-zero. All it would take is an `if` statement around the `setLatch` calls in `egs_fluence_scoring.h`.

Reply:
@ftessier no time now. When are you planning on releasing?